File: AndroidRest/src/main/java/com/android/rest/domain/Report.java
package com.android.rest.domain;


import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
public class Report {

    @Id
    @Column(unique = true)
    private Long id;

    @Column(unique = true)
    private String reportNO;


    private String bloodTest;

    private String urinTest;

    private String otherTest;

    @JsonFormat(pattern = "dd/MM/yyyy")
    private Date reportDate;

    @PrePersist
    void reportDate()
    {
        this.reportDate = new Date();
    }


    @ManyToOne
    @JoinColumn(name="user_id")
    private User user;



    @ManyToOne
    @JoinColumn(name="hospital_id")
    private Hospital hospital;

}

File: AfiyaHospital/app/src/main/java/com/example/afiyahospital/data/domain/Report.kt
package com.example.afiyahospital.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable
import java.util.*

@Entity
data class Report(
    @PrimaryKey val id:Long,
    val reportNo:String,
    val bloodTest:String,
    val urinTest: String,
    val otherTest:String,
    val reportDate: String,
    val user :String,
    val hospital :String
    ): Serializable {
}